linkml/transformers/logical_model_transformer.py

```diff
@@ -0,0 +1,710 @@
+"""
+A model transformer that transforms a schema into logical form with inheritance remaoved.
+
+Formally, this transformer generates a logical model of the schema in which inheritance
+of slots/attributes is replaced by boolean conjunctions.
+
+For example, given a slot s and two classes C and D, where C is_a D. If both C and D
+place constraints on s (for example, C may refine the range, or make more restricted
+value constraints), then an attribute is generated for C that for which the conjunction
+of constraints in both classes hold.
+
+These logical constraints are then simplified by translating to disjunctive normal form,
+and applying simplification rules.
+"""
+import logging
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Iterator, List, Union
+
+from linkml_runtime.dumpers import json_dumper
+from linkml_runtime.linkml_model import ClassDefinitionName, SchemaDefinition
+from linkml_runtime.linkml_model.meta import (
+    AnonymousSlotExpression,
+    AnonymousTypeExpression,
+    SlotDefinition,
+    SlotDefinitionName,
+    TypeDefinition,
+)
+
+from linkml.transformers.model_transformer import ModelTransformer
```

Comment:
Looks like the tests can't run because this import can't be resolved. I don't see a a `model_transformer.py` module here. Maybe that didn't get checked in?